lib/auth/storage/storage.go

```diff
@@ -233,6 +234,30 @@ func (p *ProcessStorage) WriteTeleportVersion(ctx context.Context, version *semv
 	return nil
 }
 
+func (p *ProcessStorage) rdpLicenseKey(majorVersion, minorVersion uint16, issuer, company, productID string) backend.Key {
+	return backend.NewKey("rdplicense", issuer, strconv.Itoa(int(majorVersion)), strconv.Itoa(int(minorVersion)), company, productID)
+}
+
+// WriteRDPLicense writes an RDP license to local storage.
+func (p *ProcessStorage) WriteRDPLicense(ctx context.Context, majorVersion, minorVersion uint16, issuer, company, productID string, license []byte) error {
```

Comment:
How many of these rdp licenses do we expect to handle? Should we have an expiration time? As written, they'll just accumulate in process storage with nothing ever cleaning them up.

Reply:
In the most common case, probably only one per agent.

At the high end, one for each different version of Windows that the agent connects to, so probably somewhere between 3-5.

The spec is not clear on how long they are typically valid for - it's expected that if you provide an expired license the server will just ask you to upgrade it.

I can put in a modest expiration (14 days? 30 days?) for now and adjust it after I've got the end-to-end flow working against a real server.

Reply:
Is the consequence of not having a license cert just a more involved connection process? If so, I wonder if keeping them in memory would be easier; either way, a time based expiration seems perfectly reasonable to me.

Reply:
I thought about keeping them in memory, and I might even do that for v14 to keep the v14 fix as simple as possible, but long term we should be persisting them, otherwise we consume more of our customers' licenses than necessary if the agent restarts.